app.rb

```diff
@@ -0,0 +1,84 @@
+require 'sinatra/base'
+require 'sinatra/reloader'
+require './lib/user'
+require './lib/post'
+require './lib/comment'
+require 'sinatra/flash'
+
+class Chitter < Sinatra::Base
+
+  enable :sessions
+  register Sinatra::Flash
+
+  get '/' do
+    @user = User.find_by_id(session[:user_id])
+    @post = Post.order(created_at: :desc).pluck(:peep, :created_at, :id, :user_id)
+    @comment = Comment.order(created_at: :asc)
+    erb :index
+  end
+
+  post '/add_post' do
+    new_post = Post.new(peep: params[:peep], user_id: session[:user_id])
+    new_post.save
+    redirect '/'
+  end
+
+  get '/sign-up' do
+    erb :sign_up
+  end
+
+  get '/add-comment/:id' do
+    @post_id = params[:id]
+    @post_text = Post.find_by_id(@post_id).peep
```

Comment:
Here you're calling twice `Post.find_by_id` to get the post information (first the "peep" text, then the user ID) - although I think ActiveRecord optimises for this in some way (and even though this is hidden to you by the library), it means making two SQL requests to the database - which is not ideal, as your application needs to scale. It might be better to call it only once, and then assign the different instances variables if needed:
```ruby
@post = Post.find_by_id(@post_id)
@post_text = @post.peep
@post_user_id = @post.user_id
```